tests/unit/test_web3.py

```diff
@@ -15,6 +17,74 @@ def test_sign_verify():
     assert ver is True
 
 
-def test_connect():
-    connected: bool = inst.connect("https://sepolia.base.org")
+@pytest.fixture
+def tester_provider():
+    return EthereumTesterProvider()
+
+
+@pytest.fixture
+def eth_tester(tester_provider):
+    return tester_provider.ethereum_tester
+
+
+@pytest.fixture
+def w3(tester_provider):
+    return Web3(tester_provider)
+
+
+@pytest.fixture
+def agent_registry_contract_address(w3) -> str:
+    with open(
+            "./contracts/contracts/artifacts/aregistry/AgentRegistry.abi", "r"
+    ) as abi_file:
+        abi = abi_file.read().rstrip()
+    with open(
+            "./contracts/contracts/artifacts/aregistry/AgentRegistry.bin", "r"
+    ) as bin_file:
+        bin = bin_file.read()
+    w3.eth.default_account = w3.eth.accounts[0]
+    registry = w3.eth.contract(abi=abi, bytecode=bin)
+    tx_hash = registry.constructor(
+        inst.account.address, "0x153E8ea256fDC02487882aa48A009D3573C25F99"
+    ).transact()
+    tx_receipt = w3.eth.wait_for_transaction_receipt(tx_hash)
+    return tx_receipt.contractAddress
+
+
+def test_register_agent_local(agent_registry_contract_address, w3):
+    inst.set_web3_provider(w3)
+    connected: bool = inst.connected()
     assert connected is True
+    fee = 100
+    name = "GPT"
+    tx_hash = w3.eth.send_transaction(
+        {
+            "from": w3.eth.accounts[0],
+            "to": inst.account.address,
+            "value": inst.get_gas(
+                contract_address=agent_registry_contract_address, fee=fee, name=name
+            ),
+        }
+    )
+    w3.eth.wait_for_transaction_receipt(tx_hash)
+    ok = inst.register(
+        contract_address=agent_registry_contract_address, fee=fee, name=name
+    )
+    assert ok is True
+    # try to register again
+    ok = inst.register(
+        contract_address=agent_registry_contract_address, fee=fee, name=name
+    )
+    assert ok is False
+
+# def test_register_agent_Holesky():
```

Comment:
Please, don't commit commented code

Reply:
yeah, I know this nono, I will it to readme at contracts to save contract addresses at holesky